### ISSUE TITLE
[Feature Request] A way to optimize your rom library space

### ISSUE THREAD

**Issue by yreb** [recalbox/recalbox-os#1008]
Hello

I used emulator since ten years. Here is a feature that could be very useful for recalbox: optimize game into a compatible format.

For exemple:

SMS, SNES, MD rom are classically compressed in .zip .7z  you shouls provide a way to a compress this games to the best way for your device.

This option is not very useful for that devices but could be incredible when you talk of psx iso. Many pc plugin or emulator allow some modifications to your iso.

In compression (sometimes decompression on the fly needs big CPU) 

- compress to bz2 format (cdr mooby 2 plugin for PX) 
- compress to 7z
- compress to zip
- compress to pbp or to multi pbp file

To gain some place, they also tried some exotic thing (on pc) this modification works on some andorid / linux emulator. 
- convert XA audio stream to mp3.
- Delete/compress dummy files

Other drastic option exist (wipe video data / wipe audio data)

Some times they also used index files to link the file to the compressed iso the emulator 

Other path could be explored to gain some precious place and cpu usage , convert sony video PSX files  to a newer and better compressed format.

Adding a way to do that optimizations safely and easily on your iso would be wonderful (we could gain place / cpu usage with a good compatibility) 
=> the raspberry could do it at his own rythm ( even i f the process needs one or two hours per iso)

That thing could be done for psp iso  and perhaps one day gamecube and wii iso (like the wbfs format)

**Comment by paradadf** [comment]
IMHO the Recalbox itself isn't there to convert filetypes, but to enjoy using it for playing retro games.
And for the extension support... I believe you should better open an issue upstream (([Libretro Github)](https://github.com/libretro) for every emulator you think it should support another format. Recalbox uses Retroarch and the Libretro cores (with few exceptions, but the principle is the same)  for the emulation.

**Comment by Voljega** [comment]
In 2016 a 2TB hard drive costs less than 80 euros and a pretty complete US/En PSX romset (a little less than 1100 games) takes only 500-600 Gb or so.

Uncompressing big files like PSX isos/roms on the Pi would take a long time before the game is launched, somewhere between 5 and 15 seconds, not to mention potential problem s with save files...

Given all that, I don't really see the point of your proposed modification

**Comment by yreb** [comment]
libretro is developped for a wide type of device including tablet/phone cpos. That devices are limited in storage space.

I tried many of this trick on my isos and it seems to work with some emulator whitout a huge increase in decoding time. (psx emulators are port of PC emulator)

Some of this optimizations could speed up emulation:
- convert XA audio file to mp3 could lead to an easier and hardware way to decode audio PSX game. (Less software/ More hardware)
- convert psx video file .str to an other format which is commonly hardware supported too

**Comment by Madcore** [comment]
I'm very happy with the PBP format on PSX games, and don't mind to wait 10 seconds for the game to load. Anyway, find better format and size optimizations is interesting not only on PSX but on other systems like Mega CD or Dreamcast. I remember Picodrive having MP3 support for Mega CD audio files on GP2X (also, Fusion have it too), so implementing those optimizations on other cores, not only on PSX, it's a plausible way to optimize the sdcard space.

**Comment by paradadf** [comment]
That's something to discuss with the emulator developers upstream (from libretro, the original ones or whatever), not something from Recalbox.

**Comment by wraybowling** [comment]
I agree with @Voljega's counter to this suggestion. I've been uncompressing all of my games since moving to the Raspberry Pi. It used to feel necessary when roms were being stored on 3.5" floppy disks, or CD-ROMs for my dreamcast emulator. But really, it's just much faster to boot roms that are not compressed, zips create file naming ambiguity for save states, and storage is cheap now.
